lib/quiltt_connector.dart

```diff
@@ -3,70 +3,161 @@ library quiltt_connector;
 
     Navigator.push(context, MaterialPageRoute(builder: (BuildContext context) {
-      return webViewPage.build(context);
+      return _webViewPage.build(context,
+          token: sessionToken, connectionId: config.connectionId);
     }));
   }
 }
 
 class _WebViewPage {
-  late Function(Result result) _success;
-  late Configuration _config;
-  BuildContext? _context;
-
-  _init(Configuration config, Function(Result result) success,
-      BuildContext context) {
-    _success = success;
-    _config = config;
-    _context = context;
+  late WebViewController controller;
+  late BuildContext context;
+  late QuilttConnectorConfiguration config;
+  String? token;
+  Function(Event event)? onEvent;
+  Function(Event event)? onExit;
+  Function(Event event)? onExitSuccess;
+  Function(Event event)? onExitAbort;
+  Function(Event event)? onExitError;
+
+  _init(controller, context, QuilttConnectorConfiguration config,
+      {String? token,
+      Function(Event event)? onEvent,
+      Function(Event event)? onExit,
+      Function(Event event)? onExitSuccess,
+      Function(Event event)? onExitAbort,
+      Function(Event event)? onExitError}) {
+    this.controller = controller;
+    this.context = context;
+    this.token = token;
+    this.config = config;
+    this.onEvent = onEvent;
+    this.onExit = onExit;
+    this.onExitSuccess = onExitSuccess;
+    this.onExitAbort = onExitAbort;
+    this.onExitError = onExitError;
   }
 
   _closeWebView() {
-    if (_context != null && Navigator.canPop(_context!)) {
-      Navigator.pop(_context!);
+    if (Navigator.canPop(context)) {
+      controller.clearLocalStorage();
+      Navigator.pop(context);
     }
   }
 
   _handleQuilttConnectorEvent(Uri uri) async {
+    EventMetadata eventMetadata = EventMetadata(
+      connectorId: config.connectorId,
+      connectionId: uri.queryParameters['connectionId'],
+      moveId: uri.queryParameters['moveId'],
+    );
+    String eventType = uri.host;
     switch (uri.host) {
-      case "oauthrequested":
+      case 'oauthrequested':
         var oauthUrl = Uri.decodeFull(uri.queryParameters['oauthUrl']!);
         await launchUrlString(oauthUrl, mode: LaunchMode.externalApplication);
         break;
-      case "exitsuccess":
-        _success(Result(uri.queryParameters['connectionId']));
+      case 'exitsuccess':
+        onExit?.call(Event(type: eventType, eventMetadata: eventMetadata));
+        onExitSuccess
+            ?.call(Event(type: eventType, eventMetadata: eventMetadata));
         _closeWebView();
-      case "exitabort":
-      case "exiterror":
+      case 'exitabort':
+        onExit?.call(Event(type: eventType, eventMetadata: eventMetadata));
+        onExitAbort?.call(Event(type: eventType, eventMetadata: eventMetadata));
+        _closeWebView();
+      case 'exiterror':
+        onExit?.call(Event(type: eventType, eventMetadata: eventMetadata));
+        onExitAbort?.call(Event(type: eventType, eventMetadata: eventMetadata));
```

Comment:
Based on line `130`, I'm expecting that dart needs breaks for switch statements, which means these cases will flow into each other creating unexpected event triggers.

Reply:
Fixed in https://github.com/quiltt/quiltt-connector-flutter-sdk/pull/2